From 05126f7c37b6ca692aae42fa45db01ef3e0cc1ca Mon Sep 17 00:00:00 2001
From: John Greene <jogreene@redhat.com>
Date: Mon, 8 Sep 2014 15:07:27 -0400
Subject: [ethernet] ixgbevf: remove redundant workaround

Message-id: <1410188883-6595-13-git-send-email-jogreene@redhat.com>
Patchwork-id: 92338
O-Subject: [RHEL7.1 PATCH 12/48] ixgbevf: remove redundant workaround
Bugzilla: 1091124
RH-Acked-by: Neil Horman <nhorman@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: John Linville <linville@redhat.com>

BZ: 1091124
Brew: https://brewweb.devel.redhat.com/taskinfo?taskID=7919168
Tested: In beaker. See BZ for details

This patch removes a workaround related to header split, which is redundant
because the driver does not support splitting packet headers on Rx.

Signed-off-by: Emil Tantilov <emil.s.tantilov@intel.com>
Tested-by: Phil Schmitt <phillip.j.schmitt@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
(cherry picked from commit 76b81748d43f2c60774c2703e3a9390bcc552adb)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/ixgbevf/ixgbevf_main.c b/drivers/net/ethernet/intel/ixgbevf/ixgbevf_main.c
index 3869918..ff4634f 100644
--- a/drivers/net/ethernet/intel/ixgbevf/ixgbevf_main.c
+++ b/drivers/net/ethernet/intel/ixgbevf/ixgbevf_main.c
@@ -497,15 +497,6 @@ static int ixgbevf_clean_rx_irq(struct ixgbevf_q_vector *q_vector,
   total_rx_bytes += skb->len;
   total_rx_packets++;
 
-  /*
-   * Work around issue of some types of VM to VM loop back
-   * packets not getting split correctly
-   */
-  if (staterr & IXGBE_RXD_STAT_LB) {
-   u32 header_fixup_len = skb_headlen(skb);
-   if (header_fixup_len < 14)
-    skb_push(skb, header_fixup_len);
-  }
   skb->protocol = eth_type_trans(skb, rx_ring->netdev);
 
   /* Workaround hardware that can't do proper VEPA multicast
-- 
1.7.1